### PR TITLE
bug on Refresh for Checkout pages

### PR DESCRIPTION
**Pull request by dendoesit**
## What?

See title

## Why?

Was hard to test pages if it keeps failing.

### CODE REVIEW

**Review by amandaflod — APPROVED** [review]
Good fix! 😊

File: src/client/pages/Checkout/CheckoutDetails/CheckoutDetails.tsx [hunk 1]
```diff
@@ -18,7 +18,9 @@ export const CheckoutDetails = () => {
   const { path: localePath } = useCurrentLocale()
   const { quoteCartId } = useQuoteCartIdFromUrl()
   const data = useQuoteCartData()
-
+  if (!data || data.loading) {
+    return <LoadingPage loading />
+  }
```
Comment (robinandeer) [inline src/client/pages/Checkout/CheckoutDetails/CheckoutDetails.tsx:23]: nit:
```suggestion

  if (!data || data.loading) {
    return <LoadingPage loading />
  }

```

File: src/client/pages/Checkout/CheckoutPayment/index.tsx [hunk 2]
```diff
@@ -6,15 +6,10 @@ import { CheckoutPayment } from './CheckoutPayment'
...
-    bundleVariants,
-    priceData,
-    quoteCartId,
-    mainQuote,
-    error,
-    loading,
-  } = data
+  if (!data || data.loading) {
+    return <LoadingPage loading />
+  }
```
Comment (robinandeer) [inline src/client/pages/Checkout/CheckoutPayment/index.tsx:11]: nit:

```suggestion

  if (!data || data.loading) {
    return <LoadingPage loading />
  }

```

File: src/client/utils/hooks/useQuoteCartData.ts [hunk 3]
```diff
@@ -191,7 +191,7 @@ export const useQuoteCartData = () => {
     selectedInsuranceTypes,
   )
 
-  if (!selectedQuoteBundleVariant) throw new Error('Invalid state')
+  if (!selectedQuoteBundleVariant) return
```
Comment (robinandeer) [inline src/client/utils/hooks/useQuoteCartData.ts:194]: Is this a valid state now? Or what are we representing by returning `undefined`?

Comment (dendoesit) [reply src/client/utils/hooks/useQuoteCartData.ts]: replaced this with null 👍

**Review by robinandeer — APPROVED** [review]

**Pull request merged**